c/main.c

```diff
@@ -0,0 +1,150 @@
+      case DC_EVENT_INFO:
+        printf("[Info] %s\n", message);
+        break;
+      case DC_EVENT_WARNING:
+        printf("[Warn] %s\n", message);
+        break;
+      }
+    } else if (event_type == DC_EVENT_CONFIGURE_PROGRESS) {
+
+      int progress = dc_event_get_data1_int(event);
+      char *comment = dc_event_get_data2_str(event);
+      printf("[configure-progress] %d %s\n", progress, comment);
+
+      if (progress == 0) {
+        // Failed to configure
+        printf(
+            "[BOT] configuration failed, maybe your credentials are incorect? "
+            "look for error messages above and restart the bot to try again");
+      } else if (progress == 1000) {
+        printf("[BOT] confuguration sucessfull, starting io");
+        dc_start_io(context);
+      }
+
+    } else if (event_type == DC_EVENT_MSGS_CHANGED) {
+      int chat_id = dc_event_get_data1_int(event);
+      int message_id = dc_event_get_data2_int(event);
+      printf("[msg-changed] %d %d\n", chat_id, message_id);
+
+      dc_msg_t *msg = dc_get_msg(context, message_id);
+
+      if (dc_msg_get_chat_id(msg) == DC_CHAT_ID_DEADDROP) {
+        // check if contact is new / and accept its contact request
+        printf("[BOT] accepting new contact\n");
+        int new_chat_id = dc_decide_on_contact_request(context, message_id,
+                                                       DC_DECISION_START_CHAT);
+        handle_message(context, new_chat_id, message_id);
+      }
+      dc_msg_unref(msg);
+    } else if (event_type == DC_EVENT_INCOMING_MSG) {
+      int chat_id = dc_event_get_data1_int(event);
+      int message_id = dc_event_get_data2_int(event);
+      printf("[incomming-msg] %d %d\n", chat_id, message_id);
+      handle_message(context, chat_id, message_id);
+    } else {
+      printf("[?] unhandled event of type: %d\n", event_type);
+    }
+
+    dc_event_unref(event);
+  }
+  dc_event_emitter_unref(emitter);
+}
+
+void stop_context(dc_context_t *context) {
+  dc_stop_io(context);
+  dc_stop_ongoing_process(context);
+}
+
+int main() {
+  char *addr = getenv("addr");
+  char *mailpw = getenv("mailpw");
+
+  printf("starting bot\n");
+  dc_context_t *context = dc_context_new(NULL, "deltachat-db/dc.db", NULL);
+
+  static pthread_t event_thread;
+  if (pthread_create(&event_thread, NULL, event_handler, context) != 0) {
+    printf("Event Thread creation failed\n");
+    stop_context(context);
+    return 1;
+  }
+
+  if (!dc_is_configured(context)) {
+    if (!addr) {
+      printf("you need to specify the addr enviroment variable to the bots "
+             "email address\n");
+    }
+
+    if (!mailpw) {
+      printf("you need to specify the mailpw enviroment variable to the bots "
+             "email password\n");
+    }
+    if (!addr || !mailpw) {
+      stop_context(context);
+      printf("shutting down...\n");
+      sleep(3);
+      return 1;
+    }
+
+    printf("configuring bot then after that start io to wait for messages\n");
```

Comment:
I would just say "configuring bot" here, because I/O is started when progress is 1000 in a completely different place.